Extension/src/LanguageServer/client.ts

```diff
@@ -987,6 +1076,17 @@ export class DefaultClient implements Client {
                     } else {
                         this.configuration.CompilerDefaults = compilerDefaults;
                     }
+                    // update all client configurations
+                    this.configuration.setupConfigurations();
+                    clientCount++;
```

Comment:
Does this need to be reset? I'm concerned about what happens if cpptools crashes and it might recreate a client, not sure.

Reply:
Should this code be moved the ClientCollection? The ClientColleciton is usually what deals with stuff related to multiple clients.

Reply:
@Colengms could you give your input on this?

Reply:
@sean-mcmanus  The user should not be prompted to select a trusted compiler until after all processing of configurations, as any user-specified compiler would be implicitly trusted, making the prompt unnecessary.  So, this check needs to occur after all clients are created, async initialization of the native process is complete, and all configurations have been processed by the server.  We won't know about all user-specified compilers until the native process is up and running and has processed configurations for all workspace folders, as a compile_commands.json may be used, and those are processed entirely on the native side.  Can you suggest a better place to move this check to?  The current location may actually not be sufficient, as the native process may not yet have processed compile_commands.json files yet.  One solution to that might be to make didChangeCppProperties a request instead of a notification, so that we get a response only once it's done.

Reply:
Yeah, so why can't it be moved to the end of the ClientCollection constuctor?

Reply:
Okay, I see now, the ClientCollection doesn't block for the clients to actually be ready.

Reply:
Also, it looks like it doesn't need to be reset due to "displayedSelectCompiler" being checked.